client/js/components/actionbar/time-input.js

```diff
@@ -0,0 +1,28 @@
+var React = require('react');
+var MaskedInput = require('react-maskedinput');
+
+var TimeInput = React.createClass({
+  logger: function () {},
+  render: function () {
+    return (
+      <MaskedInput
+        pattern="11:11 zm"
+        placeholder={ this.props.placeholder }
+        onClick={ this.props.handleTimeClick }
+        onChange={ this.logger }
+        formatCharacters={{
+          'z': {
```

Comment:
formatCharacters allows for custom pattern validation and transformation beyond the default options. So based on the functions below, the pattern will check to make sure z is either the letters 'a' or 'p' and then m makes sure the second letter is 'm'